params/config.go

```diff
@@ -171,6 +188,142 @@ type UpgradeConfig struct {
 	PrecompileUpgrades []PrecompileUpgrade `json:"precompileUpgrades,omitempty"`
 }
 
+func (c *UpgradeConfig) Hash() (common.Hash, error) {
+	bytes, err := c.MarshalBinary()
+	if err != nil {
+		return common.Hash{}, err
+	}
+	return crypto.Keccak256Hash(bytes), nil
+}
```

Comment:
We should probably move all of these to a separate `upgrade_config` file in the same package.